File: tests/code/corpus/langs/zig/shapes.zig
const std = @import("std");

const Circle = struct {
    radius: f64,

    pub fn area(self: Circle) f64 {
        return 3.14159 * self.radius * self.radius;
    }
};

const Kind = enum {
    round,
    square,
};

pub fn makeCircle(radius: f64) Circle {
    return Circle{ .radius = radius };
}

pub fn totalArea(shapes: []const Circle) f64 {
    var total: f64 = 0.0;
    for (shapes) |shape| {
        total += shape.area();
    }
    return total;
}

test "circle has area" {
    const c = makeCircle(1.0);
    try std.testing.expect(c.area() > 0.0);
}

File: tests/code/corpus/langs/c/shapes.c
#include <stdio.h>
#include "shapes.h"

struct Circle {
    double radius;
};

union Value {
    long i;
    double d;
};

enum Kind {
    ROUND,
    SQUARE
};

typedef struct Circle CircleAlias;

double circle_area(struct Circle *c) {
    return 3.14159 * c->radius * c->radius;
}

double total_area(struct Circle *shapes, int count) {
    double total = 0.0;
    for (int i = 0; i < count; i++) {
        total += circle_area(&shapes[i]);
    }
    return total;
}

void report(struct Circle *c) {
    printf("%f\n", circle_area(c));
}
